python/tests/unit/processes/local_runtime/test_local_step.py

```diff
@@ -0,0 +1,558 @@
+    mock_state = KernelProcessStepState(name="my_step", id="123", state=None)
+    mock_info = KernelProcessStepInfo(inner_step_type=AsyncMock(spec=type), state=mock_state, output_edges={})
+
+    step = LocalStep(
+        kernel=mock_kernel,  # type: ignore
+        step_info=mock_info,
+        event_namespace="ns",
+        step_state=mock_state,
+        factories={},
+        initialize_task=False,
+    )
+
+    # Force the initialize_step to not fill in functions.
+    async def mocked_init_step():
+        step.functions = {}
+        step.inputs = {}
+        step.initial_inputs = {}
+
+    with patch.object(
+        LocalStep,
+        "initialize_step",
+        return_value=AsyncMock(side_effect=mocked_init_step),
+    ):
+        msg = LocalMessage(
+            source_id="source",
+            destination_id="dest",
+            function_name="any_func",
+            values={"param": "value"},
+        )
+
+        with pytest.raises(ProcessFunctionNotFoundException) as exc:
+            await step.handle_message(msg)
+
+        assert "Function any_func not found in plugin my_step" in str(exc.value)
+
+
+async def test_handle_message_updates_inputs_and_invokes_function():
+    """Test that handle_message updates inputs with message values and invokes the function
+    if all parameters are provided."""
+    mock_kernel = AsyncMock(spec=Kernel)
+    mock_kernel.invoke = AsyncMock(return_value=MagicMock(value="result"))
+
+    mock_state = KernelProcessStepState(name="my_step", id="123", state=None)
+    mock_info = KernelProcessStepInfo(inner_step_type=AsyncMock(spec=type), state=mock_state, output_edges={})
+
+    # Create a function that requires one parameter
+    mock_function = AsyncMock(spec=KernelFunction)
+    mock_function.name = "func"
+    mock_function.plugin_name = "test_plugin"
+
+    mock_function.metadata = AsyncMock(spec=KernelFunctionMetadata)
+    mock_function.metadata.name = "func"
+    mock_function.metadata.plugin_name = "test_plugin"
+    mock_function.metadata.is_prompt = False
+
+    step = LocalStep(
+        kernel=mock_kernel,
+        step_info=mock_info,
+        event_namespace="ns",
+        step_state=mock_state,
+        factories={},
+        functions={
+            "func": mock_function,
+        },
+        inputs={"func": {"param": None}},
+        initial_inputs={"func": {"param": None}},
+        initialize_task=True,
+    )
+
+    with patch.object(LocalStep, "emit_event") as mock_emit_event:
+        msg = LocalMessage(
+            source_id="source",
+            destination_id="dest",
+            function_name="func",
+            values={"param": "value"},
+        )
+
+        await step.handle_message(msg)
+
+        # Function invoked with correct arguments
+        mock_kernel.invoke.assert_awaited_once()
+        mock_emit_event.assert_awaited()
+
+        # After invocation, input is reset
+        assert step.inputs["func"]["param"] is None
```

Comment:
Should we also add an assertion to check that emit_event is called with id == "func.OnResult"?